genmod/score_variants/score_variant.py

```diff
@@ -30,86 +35,111 @@ def get_plugin_score(variant, plugin_name, config_parser, csq_format=None):
     # Score is allways a number
     score = score_function.get_score(value)
     logger.debug("Score is {0} for plugin {1}".format(score, plugin_name))
-    
-    return score
 
-def get_category_score(variant, category, config_parser, csq_format=None):
-    """Return the score for a given category
+    plugin_score_min: float = score_function.score_min
+    plugin_score_max: float = score_function.score_max
+    logger.debug(f'Minmax scores for plugin {plugin_name} is ({plugin_score_min},{plugin_score_max})')
     
+    return score, plugin_score_min, plugin_score_max
+
+
+def get_category_score(variant, category, config_parser, csq_format=None) -> Tuple[int, float, float]:
+    """Return the score for a given category.
+
+       A category (such as 'allele_frequency') groups multiple plugin scores.
+       This method selects final score based on category_aggregation [Categories]
+       rule (see genmod_example.ini)
+
         Args:
             variant (dict): A variant dictionary
             category (str): A score category
             config_parser (ConfigParser): A config parser object with score functions
         Returns:
-            category_score (float): The rank score for this variant
+            category_score, sum of min and max scores for this category
     """
-    
+
     category_score = 0
-    
+
     logger.debug("Checking scores for category {0}".format(category))
     #We save all the scores for a category
     category_aggregate = config_parser.categories[category]['category_aggregation']
     category_scores = []
-    
+    category_score_mins: List[float] = []
+    category_score_maxs: List[float] = []
+
     for plugin_name in config_parser.categories[category]['plugins']:
-        category_scores.append(get_plugin_score(
-            variant = variant, 
-            plugin_name = plugin_name, 
+        plugin_score, plugin_score_min, plugin_score_max = get_plugin_score(
+            variant = variant,
+            plugin_name = plugin_name,
             config_parser = config_parser,
             csq_format = csq_format
             )
-        )
+        category_scores.append(plugin_score)
+        # Add the maximal and minimal score points that can be provided for this category
+        category_score_mins.append(plugin_score_min)
+        category_score_maxs.append(plugin_score_max)
+
+    # Provide default plugin bounds in case there were no plugins in category
+    category_score_min: float = 0.0
+    category_score_max: float = 0.0
 
     if category_scores:
-    
+
         if category_aggregate == 'max' and category_scores:
             logger.debug("Take the max score for category {0}".format(
                 category))
             category_score = max(category_scores)
+            category_score_min = min(category_score_mins)
```

Comment:
Hm, wouldn't that be the `max(category_score_mins)`, to mirror the aggregate min rule below? But, this is close enough for my fevery brain... 😆

Reply:
Yes, I was thinking to keep the range as wide as possible here, but I think it will be easier to understand the behavior of this code if the rules are consistent.

Will change this to be `max(category_score_mins)`